type-metadata/src/registry.rs

```diff
@@ -30,11 +30,7 @@ impl Registry {
 		// usage, they're assumed to be decodable by any valid decoder impl.
 		let should_ignore = match type_ident {
 			IdentKind::Custom(_) => false,
-			IdentKind::Array(_)
-			| IdentKind::Slice(_)
-			| IdentKind::Tuple(_)
-			| IdentKind::Option(_)
-			| IdentKind::Result(_) => {
+			IdentKind::Array(_) | IdentKind::Slice(_) | IdentKind::Tuple(_) => {
```

Comment:
You only do something for `Array`, `Slice` and `Tuple` here and you ignore their internals so you may also just go for a simpler `if let` construct like this:

```rust
if let IdentKind::Array(_) | IdentKind::Slice(_) | IdentKind::Tuple(_) = type_ident {
    f(self);
    return; // early return instead of saving the state in `should_ignore`
}
```